File: 160. Intersection of Two Linked Lists/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
# 1663518227
class Solution:
    def getIntersectionNode(self,headA:ListNode,headB:ListNode)->Optional[ListNode]:
        t1,t2 = headA,headB
        while t1!=t2:
            t1 = headB if t1==None else t1.next
            t2 = headA if t2==None else t2.next
        return t1

File: 160. Intersection of Two Linked Lists/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
// 1663500303
class Solution{
public:
    ListNode *getIntersectionNode(ListNode *headA,ListNode *headB){
        ListNode *t1 = headA, *t2 = headB;
        vector<ListNode*> v1,v2;
        while(t1!=nullptr){
            v1.push_back(t1);
            t1 = t1->next;
        }
        while(t2!=nullptr){
            v2.push_back(t2);
            t2 = t2->next;
        }
        for(auto i:v1){
            for(auto j:v2){
                if(i==j){return i;}
            }
        }
        return nullptr;
    }
};
